packages/flutter_tools/lib/src/bundle.dart

```diff
@@ -122,22 +122,26 @@ Future<void> buildWithAssemble({
   final Target target = buildMode == BuildMode.debug
     ? const CopyFlutterBundle()
     : const ReleaseCopyFlutterBundle();
-  final BuildResult result = await buildSystem.build(target, environment);
+  try {
+    final BuildResult result = await buildSystem.build(target, environment);
 
-  if (!result.success) {
-    for (ExceptionMeasurement measurement in result.exceptions.values) {
-      printError(measurement.exception.toString());
-      printError(measurement.stackTrace.toString());
+    if (!result.success) {
+      for (ExceptionMeasurement measurement in result.exceptions.values) {
+        printError(measurement.exception.toString());
+        printError(measurement.stackTrace.toString());
+      }
+      throwToolExit('Failed to build bundle.');
```

Comment:
Does this not get caught?

Reply:
This only catches errors which happen during the build target's execution, and not permission errors hit during the build system itself

Reply:
I guess I'm asking if throwing a tool exit here will get caught and wrapped in a second tool exit?

Reply:
oh...hmmm. I"m being silly